gems/pending/fs/xfs/allocation_group.rb

```diff
@@ -115,8 +115,8 @@ def initialize(stream, _agno, sb)
       @stream.seek(AG_INODEINFO_SIZE, IO::SEEK_CUR)
       @agfl                   = AG_FREELIST.decode(@stream.read(AG_FREELIST_SIZE))
       @stream.seek(-(AG_FREESPACE_SIZE + AG_INODEINFO_SIZE + AG_FREELIST_SIZE))
-      @allocation_group_block = MemoryBuffer.create(sb['block_size'])
-      @allocation_group_block = @stream.read(sb['block_size'])
+      @allocation_group_block                      = MemoryBuffer.create(sb['block_size'])
+      @allocation_group_block[0, sb['block_size']] = @stream.read(sb['block_size'])
```

Comment:
Ah ok, the creation of memory buffers are typically just for passing into a C-api that will fill in the buffer for us.